src/monster.cpp

```diff
@@ -162,6 +162,17 @@ void monster::poly(mtype *t)
     faction = t->default_faction;
 }
 
+void monster::update_check(){
+    if (type->upgrade_group != "NULL"){
+        int time_passed = calendar::turn.get_turn()/ DAYS(1);
```

Comment:
Time passed should be time since the last upgrade check, not time since game start, otherwise you can't reasonably do multi-level upgrades.